src/components/Rebuild/Rebuild.js

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2019 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { Component } from 'react';
+import { Button } from 'carbon-components-react';
+import Restart from '@carbon/icons-react/lib/restart/32';
+import { rebuildPipelineRun } from '../../api';
+import './Rebuild.scss';
+
+export class Rebuild extends Component {
+  handleRebuild = event => {
+    if (event) {
+      event.preventDefault();
+    }
+
+    /* eslint no-useless-rename: "error" */
+    const pipelinerunname = this.props.runName;
+    /* eslint no-useless-rename: "error" */
+    const pipelineName = this.props.pipelineRun.spec.pipelineRef.name;
+    const { namespace } = this.props.pipelineRun.metadata.namespace; // was namespaceToUse and OK
+    /* eslint no-useless-rename: "error" */
+    const payload = {
+      /* eslint no-useless-rename: "error" */
+      pipelinerunname
+    };
+
+    const promise = rebuildPipelineRun(namespace, payload);
+
+    promise
+      .then(headers => {
+        const logsURL = headers.get('Content-Location');
+        const newPipelineRunName = logsURL.substring(
+          logsURL.lastIndexOf('/') + 1
+        );
+        const finalURL = `/namespaces/${namespace}/pipelines/${pipelineName}/runs/${newPipelineRunName}`;
+        this.props.setShowRebuildNotification(true);
```

Comment:
Rather than setting these individually can we just set a single object? Something like this:

```js
this.props.showRebuildNotification({
  message: 'Rebuilt run successfully',
  kind: 'success',
  logsURL
});
```

In PipelineRun you would `setState({ rebuildNotification })` with this object as the value.

Reply:
Great idea, trying

Reply:
Done, this was a nice refactor, thank you